[K9VULN-11161] Add documentation for CI/CD in CSM (#34137)
DataDog/documentation: content/en/security/cloud_security_management/setup/ci_cd/_index.md

## Authentication

To upload scan results to Datadog, configure the following environment variables in your CI pipeline:

| Name         | Description                                                                                                                | Required | Default         |
|--------------|----------------------------------------------------------------------------------------------------------------------------|----------|-----------------|
| `DD_API_KEY` | Your Datadog API key. This key is created by your [Datadog organization][4] and should be stored as a secret.            | Yes      |                 |
| `DD_APP_KEY` | Your Datadog application key. This key, created by your [Datadog organization][5], should include the `appsec_vm_read` scope and be stored as a secret.    | Yes      |                 |
| `DD_SITE`    | The [Datadog site][6] to send information to. Your Datadog site is {{< region-param key="dd_site" code="true" >}}.       | No       | `datadoghq.com` |

<div class="alert alert-info">
Store your API and application keys as secrets in your CI/CD platform to protect sensitive credentials.
</div>

> ## Setup instructions

Reviewer: ```suggestion
```
This heading seems redundant with the "get started" heading. I'll suggest moving this information to the next section, with the exception of the promise of stuff coming soon - we really try to avoid making those promises in the docs, because plans can change so easily.

<div class="alert alert-info">Pre-built integrations for specific CI providers (GitHub Actions, GitLab CI/CD, Azure DevOps, etc.) are coming soon. For now, use the customizable script approach below, which works with any CI/CD platform.</div>

Container image scanning works with all major CI/CD platforms including:
- GitHub Actions
- GitLab CI/CD
- Azure DevOps
- Other CI providers that can execute shell scripts

The customizable script approach gives you full control over when and how scans are executed in your pipelines.

## Installation and scans

> <div class="alert alert-info">The Datadog Security CLI is in Preview and available to install from Datadog package repositories.</div>

Reviewer: ```suggestion
{{< callout url="#" btn_hidden="true" header="Join the Preview" >}}
The Datadog Security CLI is in Preview and available to install from Datadog package repositories.
{{< /callout >}}
```

The Datadog Security CLI can be installed on Debian/Ubuntu, Red Hat/CentOS, and macOS systems. Choose your installation method below.

### Install the Datadog Security CLI

{{< tabs >}}
{{% tab "Debian/Ubuntu" %}}

### Install from package repository

```bash
# Import Datadog APT signing key
DD_APT_KEY_URL="https://keys.datadoghq.com/DATADOG_APT_KEY_CURRENT.public"
curl -fsSL "$DD_APT_KEY_URL" | sudo gpg --dearmor -o /usr/share/keyrings/datadog-archive-keyring.gpg

# Add Datadog repository
echo "deb [signed-by=/usr/share/keyrings/datadog-archive-keyring.gpg] https://apt.datadoghq.com/ stable datadog-security-cli" \
| sudo tee /etc/apt/sources.list.d/datadog-security-cli.list

# Update package list and install
sudo apt update
sudo apt install datadog-security-cli
```

{{% /tab %}}
{{% tab "Red Hat/CentOS" %}}

> ### Install from package repository

Reviewer: ```suggestion
#### Install from package repository
```

```bash
# Import Datadog RPM signing key
sudo rpm --import https://keys.datadoghq.com/DATADOG_RPM_KEY_CURRENT.public

# Add Datadog repository
sudo tee /etc/yum.repos.d/datadog-security-cli.repo > /dev/null <<'EOF'
[datadog-security-cli]
name=Datadog Security CLI
baseurl=https://yum.datadoghq.com/stable/datadog-security-cli/\$basearch/
enabled=1
gpgcheck=1
gpgkey=https://keys.datadoghq.com/DATADOG_RPM_KEY_CURRENT.public
repo_gpgcheck=1
EOF

# Install the CLI
sudo yum install datadog-security-cli
```

{{% /tab %}}
{{% tab "macOS" %}}

### Install with Homebrew

```bash
# Install via Homebrew
brew install --cask datadog-security-cli
```

{{% /tab %}}
{{< /tabs >}}

> ### Run your first scan

Reviewer: ```suggestion
#### Run your first scan
```

After installing the Datadog Security CLI, configure your Datadog credentials and scan a container image:

```bash
# Configure Datadog credentials
export DD_API_KEY=<your_api_key>
export DD_APP_KEY=<your_app_key>
export DD_SITE={{< region-param key="dd_site" >}}

# Scan your container image
datadog-security-cli image myimage:tag
```

The CLI outputs scan results directly to your terminal, showing:
- Image information (name, digest, operating system)
- Total number of vulnerabilities found
- Severity breakdown (Critical, High, Medium, Low)
- Detailed table of vulnerabilities with CVE IDs, affected packages, and available fixes

{{< img src="security/vulnerabilities/csm-vm-cli-output.png" alt="Datadog Security CLI output showing vulnerability scan results for a container image" style="width:100%;" >}}

You can also view your scan results in Datadog's [Cloud Security Vulnerabilities][3] explorer.

## Run local scans during development

For faster iteration before committing to CI, install the Datadog Security CLI locally using the same installation methods described in the [installation section](#install-the-datadog-security-cli) above.

> ### Local scanning without persisting results

Reviewer: ```suggestion
### Scan locally without persisting results
```

- **Filter by resource type**: View vulnerabilities specific to container images scanned in CI/CD
- **Prioritize by severity**: Focus on critical and high-severity vulnerabilities first
- **Track remediation**: Assign vulnerabilities to team members and track resolution
- **Set up notifications**: Get alerted when new critical vulnerabilities are detected

{{< img src="security/vulnerabilities/csm-vm-explorer-actionability-2.png" alt="The Cloud Security Vulnerabilities Findings page" width="100%">}}

## Troubleshooting

### Authentication errors

If you encounter authentication errors:
1. Verify your `DD_API_KEY` and `DD_APP_KEY` are correctly set
2. Ensure the application key has the `appsec_vm_read` scope
3. Check that your `DD_SITE` matches your Datadog organization's site

### Image not found errors

If the CLI cannot find your image:
1. Verify the image exists locally: `docker images`
2. Use the full image name including registry if applicable
3. Ensure the image is built before scanning

### Network connectivity issues

If scans fail due to network issues:
1. Verify your CI environment can reach the Datadog site
2. Check for proxy or firewall restrictions

> 3. Ensure outbound HTTPS connections are allowed

Reviewer: ```suggestion
1. Verify your CI environment can reach the Datadog site.
2. Check for proxy or firewall restrictions.
3. Ensure outbound HTTPS connections are allowed.
```

{{< img src="security/vulnerabilities/csm-vm-cli-output.png" alt="Datadog Security CLI output showing vulnerability scan results for a container image" style="width:100%;" >}}

### View scan results

After running your first scan, results appear on the [Cloud Security Vulnerabilities][3] page within minutes. You can:

- **Filter by resource type**: View vulnerabilities specific to container images scanned in CI/CD
- **Prioritize by severity**: Focus on critical and high-severity vulnerabilities first
- **Track remediation**: Assign vulnerabilities to team members and track resolution
- **Set up notifications**: Get alerted when new critical vulnerabilities are detected

{{< img src="security/vulnerabilities/csm-vm-explorer-actionability-2.png" alt="The Cloud Security Vulnerabilities Findings page" width="100%">}}

> ## Run local scans during development

Reviewer: ```suggestion
### Run local scans during development
```

For faster iteration before committing to CI, install the Datadog Security CLI locally using the same installation methods described in the [installation section](#install-the-datadog-security-cli) above.

> ### Scan locally without persisting results

Reviewer: ```suggestion
#### Scan locally without persisting results
```
